x/incentives/keeper/msg_server.go

```diff
@@ -24,7 +24,12 @@ var _ types.MsgServer = msgServer{}
 
 func (server msgServer) CreatePot(goCtx context.Context, msg *types.MsgCreatePot) (*types.MsgCreatePotResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	potID, err := server.keeper.CreatePot(ctx, msg.IsPerpetual, msg.Owner, msg.Coins, msg.DistributeTo, msg.StartTime, msg.NumEpochsPaidOver)
+	owner, err := sdk.AccAddressFromBech32(msg.Owner)
```

Comment:
Do we really have to follow this design? This is just straight up a mis-design of the SDK imo.

Why would you have all nodes on the network perform bech32 decoding in the state machine, that should just be the responsibility of the tx creator.

Reply:
Here's the context:  https://github.com/cosmos/cosmos-sdk/issues/7195

I don't fully understand it, but I'd rather not stray from what the SDK prescribes rn

Reply:
This is pretty unfortunate.

Agreed on tabling this for now then. We have bigger impact things we can pursue for chain scaling #post-launch. This is pretty sad though =/